core/unix/signal.c

```diff
@@ -4807,6 +4807,28 @@ master_signal_handler_C(byte *xsp)
         return;
     }
 #endif
+
+    /* We are dropping asynchronous signals during detach. The thread may already have
+     * lost its TLS (xref i#3535). A safe read may result into a race affecting
+     * asynchronous non-alarm signals between delivering the SIGSEGV and restoring the
+     * app's signal handlers.
```

Comment:
I found this confusing: I would suggest sthg like "A safe read may result in a crash if DR's SIGSEGV handler is removed before the safe read's SIGSEGV is delivered."

Reply:
Sure.